### ISSUE TITLE
Difference in compilations results between AWS for logging.

### ISSUE THREAD

**Issue by Matthewar** [HPCE/hpce-2017-cw5#52]
Hi,

When compiling on my machine, I get no errors, however in the automated runs the arguments to the log functions are throwing errors - is there some setting that is different between these so they can be diagnosed on the local machine?

**Comment by m8pple** [comment]
I'm not sure what you mean - the compilation settings should be in the log file? Do you
mean on your own AWS instance (in which case you should be able to do the same
command as in the log (which is just using your makefile)), or on your laptop?

My guess is that if you're getting errors with the logging, you're trying to pass something
through that is a class, rather than a base C type. e.g. you're trying to log a `std::string`,
but you need to call `.c_str()` to pass it to log (as log behaves like printf). Not sure why
that would compile on other machines though.

**Comment by Matthewar** [comment]
Yes, I was trying to pass it a c str instead but the bit I was confused about was that it let me compile it with no issues on my machine (although I got some weird outputs when it printed).
On the AWS instance it wouldn't compile
